src/OneExplorer/OneExplorer.ts

```diff
@@ -185,11 +185,26 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
       warningMessage = 'WARNING: Renaming may result in some unexpected changes on the tree view.';
     }
 
+    const validateInputPath = (newname: string): string|undefined => {
+      const oldpath = oneNode.node.path;
+      const dirpath = path.dirname(oneNode.node.uri.fsPath);
+      const newpath: string = `${dirpath}/${newname}`;
```

Comment:
(optional) how about using `path.join` for future when we support Windows? 😸 

```suggestion
      const newpath: string = path.join(${dirpath}, ${newname});
```